wp-content/plugins/hurumap/src/App.js

```diff
@@ -270,84 +298,90 @@ function App() {
+                          order: form.values.sections.length,
                           published: false
                         })
                       }
                     >
                       Add Section
                     </Button>
-                    {/* <Actions
-                      actions={[
-                        {
-                          label: 'Publish Selected',
-                          value: 'publish:selected'
-                        },
-                        { label: 'Publish All', value: 'publish:all' },
-                        { label: 'Delete Selected', value: 'delete:selected' },
-                        { label: 'Delete All', value: 'delete:all' }
-                      ]}
-                    /> */}
-                    <Grid container direction="row" spacing={1}>
-                      {form.values.sections.map(section => (
-                        <Grid key={section.id} item xs={12} md={4}>
-                          <React.Suspense fallback={<div>Loading...</div>}>
-                            <ChartsSection
-                              section={section}
-                              onChange={changes => {
-                                const updatedSection = Object.assign(
-                                  section,
-                                  changes
-                                );
-                                arrayHelper.replace(
-                                  form.values.sections.indexOf(section),
-                                  updatedSection
-                                );
-                                updateOrCreateChartsSection(updatedSection);
-                              }}
-                              onAddChart={chartId => {
-                                let chart = form.values.hurumapCharts.find(
-                                  c => c.id === chartId
-                                );
-                                if (chart) {
-                                  handleUpdateHurumapChart(form, chart, {
-                                    section: section.id
-                                  });
-                                } else {
-                                  chart = form.values.flourishCharts.find(
+                    <Grid container direction="column" spacing={2}>
+                      {form.values.sections
+                        .sort((a, b) => (a.order >= b.order ? 1 : -1))
+                        .map((section, q) => (
+                          <Grid key={section.id} item xs={12} md={6}>
+                            <React.Suspense fallback={<div>Loading...</div>}>
+                              <ChartsSection
+                                section={Object.assign(section, { order: q })}
+                                sectionsCount={form.values.sections.length}
+                                onMove={position => {
+                                  if (position === 'up') {
```

Comment:
Can't the movement check and array swapping be done in the `handleXXXChartsSection` functions? i.e. include `movement` and `arrayHelper` as parameters to these function.

Reply:
:+1: